pkg/totalaccountwatcher/totalaccountwatcher.go

```diff
@@ -125,3 +143,39 @@ func TotalAwsAccounts() (int, error) {
 
 	return accountTotal, nil
 }
+
+// AccountsCanBeCreated returns whether we can create accounts or not
+func (s *totalAccountWatcher) AccountsCanBeCreated() bool {
+	return s.accountsCanBeCreated
+}
+
+// accountLimitReached returns True if our account limit is reached or False if the account limit is not reached and we can create accounts.
+func accountLimitReached(kubeClient client.Client, log logr.Logger, currentAccounts int) (bool, error) {
+	limit, err := getAwsAccountLimit(kubeClient)
+	if err != nil {
+		log.Error(err, "There was an error getting the limits.  Using the default value.")
+		return true, err
+	}
+	return currentAccounts >= limit, err
+}
+
+// getAwsAccountLimit gets the limit from the ConfigMap or on error returns a default value.
+func getAwsAccountLimit(kubeClient client.Client) (int, error) {
```

Comment:
It this point you could probably fold this into `accountLimitReached`, which would let you do away with some of the awkwardness of the return values (recall the whole `-1` thing was a lesser evil, but still evil).

Reply:
I think it still makes sense here to return the negative value with an error and keep this functionality separate, more as a logical separation than anything, since the tests only test the public functions and not the private ones.